packages/medusa/src/commands/utils/get-migrations.js

```diff
@@ -154,3 +190,66 @@ export const getEnabledMigrations = (migrationDirs, isFlagEnabled) => {
     })
     .filter(Boolean)
 }
+
+export const getModuleMigrations = (configModule, isFlagEnabled) => {
+  const loadedModules = getInternalModules(configModule)
+
+  const allModules = []
+
+  for (const loadedModule of loadedModules) {
+    const mod = loadedModule.loadedModule
+
+    const isolatedMigrations = {}
+    const moduleMigrations = (mod.migrations ?? [])
+      .map((migrations) => {
+        const all = []
+        for (const migration of Object.values(migrations)) {
+          // TODO: revisit how Modules export their migration entrypoints up/down
+          if (["up", "down"].includes(migration.name)) {
+            isolatedMigrations[migration.name] = migration
+          } else if (
+            typeof migration.featureFlag === "undefined" ||
+            isFlagEnabled(migration.featureFlag)
+          ) {
+            all.push(migration)
+          }
+        }
+        return all
+      })
+      .flat()
+
+    allModules.push({
+      moduleDeclaration: loadedModule.moduleDeclaration,
+      models: mod.models ?? [],
+      internalMigrations: moduleMigrations,
+      externalMigrations: isolatedMigrations,
```

Comment:
Internal migrations are for internal modules using shared resources, and external is for internal modules using isolated resources (maybe should be rename to isolatedMigrations).
The former is not being used yet as there is a TODO flagged in the code, that we have to normalize how module's migration can be executed from the Core

Reply:
Ah yeah - I realise my comment was a bit off. I was not asking about their purpose of them (since that is clear from their name), but just to flag that I couldn't see them being used, so I wanted to make sure that is the case.

Thanks for explaining!